tests/e2e/velero_helpers.go

```diff
@@ -247,3 +248,58 @@ func (v *veleroCustomResource) IsDeleted() wait.ConditionFunc {
 		return false, err
 	}
 }
+
+//check if bsl matches the spec
+func doesBSLExist(namespace string, bsl velero.BackupStorageLocationSpec, spec *oadpv1alpha1.VeleroSpec) wait.ConditionFunc {
+	return func() (bool, error) {
+		if len(spec.BackupStorageLocations) > 0 {
+			for _, b := range spec.BackupStorageLocations {
+				if b.Provider == bsl.Provider {
+					if !reflect.DeepEqual(bsl, b) {
+						return false, errors.New("given Velero bsl does not match the deployed velero bsl")
+					}
+					return true, nil
+
+				}
+			}
+		}
+		return false, errors.New("no backup storage location configured. Expected BSL to be configured")
+
+	}
+}
+
+//check if vsl matches the spec
+func doesVSLExist(namespace string, vslspec velero.VolumeSnapshotLocationSpec, spec *oadpv1alpha1.VeleroSpec) wait.ConditionFunc {
+	return func() (bool, error) {
+
+		if len(spec.VolumeSnapshotLocations) > 0 {
```

Comment:
NIT: if the length of VSLs = 0, return an error. Then you don't have to indent everything else